### PR TITLE
Add network policy configuration

### LINKED ISSUE #238: Add other network profile parameters to AKS build
currently the AKS module does not reference the ServiceIP CIDR, DNS Service IP, DockerBridgeIP and Network Policy options of the AKS network profile in the AKS provider. This is important to allow for control as these default to a CIDR and IP set that usually overlaps with large enterprise Route tables and can cause hybrid connectivity issues, and also does not expose the ability to enable Calico Network Policy.

### LINKED ISSUE #293: CI tests with external modules don't test PR code
**Describe the bug:**
When a PR happens, Azure DevOps will properly pull the effective code when the PR is merged. 
 However, the Terraform tests will automatically pull the version of the module code which is currently in the master branch in the GitHub repo, effectively not testing any code changes which would impact the test from working.

**To Reproduce:**
Make a breaking change in one of the core `/cluster/azure/aks` folders and initiate a PR.  The PR will pass since it is simply testing the upstream code.

**Expected behavior:**
CI tests the effective local module code with the PR using `source = "../.."` path for the modules instead of the github.com/Microsoft/bedrock reference.

**Additional context:**
An easy workaround would be the ability to pass in a variable to the `source`, but this is a problem which has been around since 2015 with Terraform and doesn't appear to be any "feature" coming to make this work.  [See this issue ](https://github.com/hashicorp/terraform/issues/18063) and related links from there if you care to know more.  

Open to suggestions, but likely a solution will have to do a string replacement of all instances of `github.com/Microsoft/bedrock/cluster` in the environment with `../..` before the tests happen.

### PR DESCRIPTION
**Pull request by dtzar**
- Enables ability to specify `calico` for network policy. Close #238 
- Related to #305 spike investigate AKS network policy
- Fixes #293 allowing CI tests to actually test local module code versus GitHub
- Adds the ability to recreate the kube config file via `kubeconfig_recreate` for simple and single-keyvault environments again
- Updates azure provider from 1.23 --> 1.25

### CODE REVIEW

**Comment by jmspring** [comment]
@dtzar - shouldn't the policy be reflected in the more complicated environments (single-keyvault/multi) as well?

**Comment by dtzar** [comment]
Makes sense @jmspring - wanted to get the tests and visibility going.  I can add shortly to the other environments.

File: cluster/environments/azure-simple/terraform.tfvars [hunk 1]
```diff
@@ -16,3 +16,4 @@ vnet_name = "<vnet name>"
 # gitops_url_branch = "release-123"
 # gitops_poll_interval = "30s"
 # gitops_path = "prod"
+# network_policy = "calico"
```
Comment (evanlouie) [inline cluster/environments/azure-simple/terraform.tfvars]: newline please!

Comment (dtzar) [reply cluster/environments/azure-simple/terraform.tfvars]: fixed

**Comment by dtzar** [comment]
@jmspring I believe this PR is good to go.  IMO it's ok to not update the multi-cluster environment every time with every option available since this will be cumbersome to maintain.  The other two environments have policy enabled.

**Pull request merged**